components/tree/private/branch.jsx

```diff
@@ -173,7 +172,7 @@ const handleKeyDown = (event, props) => {
 };
 
 const handleFocus = (event, props) => {
-	if (!props.focusedNodeIndex) {
+	if (!props.focusedNodeIndex && event.target === event.currentTarget) {
```

Comment:
This selects the node upon first focus, but only do it if focusing the `<li>`.  When focusing the button, we allow it to go through so that the branch expand handler works.